hedera-mirror-web3/src/main/java/com/hedera/services/store/tokens/HederaTokenStore.java

```
@@ -0,0 +1,463 @@
+import com.hedera.services.store.models.NftId;
+import com.hedera.services.store.models.Token;
+import com.hedera.services.store.models.TokenRelationship;
+import com.hedera.services.store.models.UniqueToken;
+import com.hedera.services.txns.validation.ContextOptionValidator;
+import com.hederahashgraph.api.proto.java.AccountID;
+import com.hederahashgraph.api.proto.java.ResponseCodeEnum;
+import com.hederahashgraph.api.proto.java.TokenID;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.apache.commons.lang3.tuple.Pair;
+
+/** Provides a managing store for arbitrary tokens. */
+@Singleton
+public class HederaTokenStore {
+
+    static final TokenID NO_PENDING_ID = TokenID.getDefaultInstance();
+    private final ContextOptionValidator validator;
+    private final MirrorNodeEvmProperties mirrorNodeEvmProperties;
+    private final StackedStateFrames<Object> stackedStateFrames;
+    TokenID MISSING_TOKEN = TokenID.getDefaultInstance();
+    TokenID pendingId = NO_PENDING_ID;
+    Token pendingCreation;
+
+    @Inject
+    public HederaTokenStore(
+            final ContextOptionValidator validator,
+            final MirrorNodeEvmProperties mirrorNodeEvmProperties,
+            final StackedStateFrames<Object> stackedStateFrames) {
+        this.validator = validator;
+        this.mirrorNodeEvmProperties = mirrorNodeEvmProperties;
+        this.stackedStateFrames = stackedStateFrames;
+    }
+
+    public static Pair<AccountID, TokenID> asTokenRel(AccountID account, TokenID token) {
+        return Pair.of(account, token);
+    }
+
+    // Copied from HederaLedger
+    public ResponseCodeEnum usabilityOf(final AccountID id) {
+        try {
+            final var topFrame = stackedStateFrames.top();
+            final var accountAccessor = topFrame.getAccessor(Account.class);
+            final var account = accountAccessor.get(id).orElseThrow(); // TODO: Change
```

When we start using the Store interface, fetching of different models would throw exception internally if we pass a suitable flag - OnMissing.THROW.

Here, however, we wouldn't want to throw an exception but in case the account is missing we can return `INVALID_ACCOUNT_ID`, so that we mimic the services validation workflow. We should aim to return proper `ResponseCodeEnum` values.

This deviation from the original code base, should be documented with a javadoc upon this method.